src/Toolkit/Toolkit/UI/Controls/SearchView/LocatorSearchSource.cs

```diff
@@ -146,12 +171,36 @@ public LocatorSearchSource(LocatorTask locator)
         {
             Locator = locator;
 
-            _loadTask = EnsureLoaded();
+            RefreshDisplayName();
+
+            _loadTask = new Lazy<Task>(EnsureLoaded);
+        }
+
+        private void RefreshDisplayName()
+        {
+            if (_displayNameSetExternally)
+                return;
+
+            if (Locator?.LocatorInfo is LocatorInfo info)
+            {
+                // Locators from online services have descriptions but not names.
+                if (!string.IsNullOrWhiteSpace(info.Name) && info.Name != Locator.Uri?.ToString())
+                    _displayName = info.Name;
+                else if (!string.IsNullOrWhiteSpace(info.Description))
+                    _displayName = info.Description;
+                else
+                    _displayName = string.Empty;
```

Comment:
Do we want it to fall back to empty? Wasn't the old code falling back to "Locator" ?